src/app/(proper_react)/redesign/(authenticated)/user/dashboard/fix/high-risk-data-breaches/HighRiskBreachLayout.tsx

```diff
@@ -57,8 +64,59 @@ export function HighRiskBreachLayout(props: HighRiskBreachLayoutProps) {
   // The non-null assertion here should be safe since we already did this check
   // in `./[type]/page.tsx`:
   const { title, illustration, content, exposedData, type } = pageData!;
-  const hasBreaches = type !== "none";
+  const isHighRiskBreachesStep = type !== "none";
   const isStepDone = type === "done";
+  const hasExposedData = exposedData.length;
+
+  // TODO: Write unit tests MNTOR-2560
+  /* c8 ignore start */
+  const handlePrimaryButtonPress = async () => {
+    const highRiskBreachClasses: Record<
+      HighRiskBreachTypes,
+      (typeof HighRiskDataTypes)[keyof typeof HighRiskDataTypes] | null
+    > = {
+      ssn: HighRiskDataTypes.SSN,
+      "credit-card": HighRiskDataTypes.CreditCard,
+      "bank-account": HighRiskDataTypes.BankAccount,
+      pin: HighRiskDataTypes.PIN,
+      none: null,
+      done: null,
+    };
+
+    const dataType = highRiskBreachClasses[type];
+    if (!dataType || !hasExposedData || isResolving) {
+      return;
+    }
+
+    setIsResolving(true);
+    try {
+      const response = await fetch("/api/v1/user/breaches/bulk-resolve", {
+        method: "PUT",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({ dataType }),
+      });
+
+      const result = await response.json();
+      if (!result?.success) {
+        throw new Error(
+          `Could not resolve breach data class of type: ${props.type}`,
+        );
+      }
+
+      const nextRoute = getIsNextStepSectionFromPathname({
```

Comment:
Check if the next breach resolution step is within the same section or if the user would be guided to the next one. If the user gets guided to the next section that means that we just resolved the last breach in the current one and we show them the celebration view.